src/patients/related-persons/AddRelatedPersonModal.tsx

```diff
@@ -50,9 +55,13 @@ const AddRelatedPersonModal = (props: Props) => {
               placeholder={t('patient.relatedPerson')}
               onChange={onPatientSelect}
               onSearch={async (query: string) => PatientRepository.search(query)}
-              renderMenuItemChildren={(patient: Patient) => (
-                <div>{`${patient.fullName} (${patient.code})`}</div>
-              )}
+              renderMenuItemChildren={(patients: Patient) => {
+                if (patientCode() === patients.code) {
```

Comment:
Although `code` is a unique identifier, I think it would be better to compare patient ids.